opensearch-observability/src/main/kotlin/org/opensearch/observability/model/Application.kt

```diff
@@ -0,0 +1,262 @@
+ *       "Purchase.source"
+ *   ],
+ *   "availability_levels": [
+ *       {
+ *           "label": "Unavailable",
+ *           "color": "#D36086",
+ *           "condition": "when errorRate() is above or equal to 2%",
+ *           "order": "0",
+ *       }
+ *   ],
+ * }
+ * }</pre>
+ */
+
+internal data class Application(
+    val name: String?,
+    val description: String?,
+    val base_query: String?,
+    val services_entities: List<String>,
+    val trace_groups: List<String>,
+    val availability_levels: List<AvailabilityLevel>
+) : BaseObjectData {
+
+    internal companion object {
+        private val log by logger(SavedQuery::class.java)
+        private const val NAME_TAG = "name"
+        private const val DESCRIPTION_TAG = "description"
+        private const val BASE_QUERY_TAG = "base_query"
+        private const val SERVICES_ENTITIES_TAG = "services_entities"
+        private const val TRACE_GROUPS_TAG = "trace_groups"
+        private const val AVAILABILITY_LEVELS_TAG = "availability_levels"
+
+        /**
+         * reader to create instance of class from writable.
+         */
+        val reader = Writeable.Reader { Application(it) }
+
+        /**
+         * Parser to parse xContent
+         */
+        val xParser = XParser { parse(it) }
+
+        /**
+         * Parse the data from parser and create ObservabilityObject object
+         * @param parser data referenced at parser
+         * @return created ObservabilityObject object
+         */
+        fun parse(parser: XContentParser): Application {
+            var name: String? = null
+            var description: String? = null
+            var base_query: String? = null
+            var services_entities: List<String> = listOf()
+            var trace_groups: List<String> = listOf()
+            var availability_levels: List<AvailabilityLevel> = null
+            XContentParserUtils.ensureExpectedToken(
+                XContentParser.Token.START_OBJECT,
+                parser.currentToken(),
+                parser
+            )
+            while (XContentParser.Token.END_OBJECT != parser.nextToken()) {
+                val fieldName = parser.currentName()
+                parser.nextToken()
+                when (fieldName) {
+                    NAME_TAG -> name = parser.text()
+                    DESCRIPTION_TAG -> description = parser.text()
+                    BASE_QUERY_TAG -> base_query = parser.text()
+                    SERVICES_ENTITIES_TAG -> services_entities = parser.stringList()
+                    TRACE_GROUPS_TAG -> trace_groups = parser.stringList()
+                    AVAILABILITY_LEVELS_TAG -> availability_levels = AvailabilityLevel.parse(parser)
+                    else -> {
+                        parser.skipChildren()
+                        log.info("$LOG_PREFIX:Application Skipping Unknown field $fieldName")
+                    }
+                }
+            }
+            // Why is this not necessary?
```

Comment:
if you add the checks, requests need to have all fields present (even if you are just updating one field in the object)